ngraph/core/src/op/asinh.cpp

```diff
@@ -7,21 +7,42 @@
 
 #include "itt.hpp"
 #include "ngraph/op/asinh.hpp"
+#include "ngraph/op/util/elementwise_args.hpp"
 #include "ngraph/runtime/host_tensor.hpp"
 #include "ngraph/runtime/reference/asinh.hpp"
 #include "ngraph/type/element_type.hpp"
 
 using namespace std;
 using namespace ngraph;
 
-constexpr NodeTypeInfo op::v3::Asinh::type_info;
+NGRAPH_RTTI_DEFINITION(op::v3::Asinh, "Asinh", 3, util::UnaryElementwiseArithmetic);
 
 op::v3::Asinh::Asinh(const Output<Node>& arg)
     : UnaryElementwiseArithmetic(arg)
 {
     constructor_validate_and_infer_types();
 }
 
+void op::v3::Asinh::validate_and_infer_types()
+{
+    NGRAPH_OP_SCOPE(v3_Asinh_validate_and_infer_types);
+
+    NODE_VALIDATION_CHECK(
+        this, get_input_size() == 1, "Argument has ", get_input_size(), " outputs (1 expected).");
+
+    element::Type element_type = get_input_element_type(0);
+    PartialShape pshape = get_input_partial_shape(0);
+
+    NODE_VALIDATION_CHECK(this,
+                          element_type.is_dynamic() || element_type.is_real(),
```

Comment:
In the spec we don't have any limitations for element type.

Reply:
the spec (even before re-format) says: T is any floating point type.

Do you mean that spec is wrong, T can be any numeric type?

Reply:
Changed to any numeric type